extra_data/exceptions.py

```diff
@@ -37,3 +37,15 @@ def __str__(self):
             "because you have used .union() to combine data. Please retrieve "
             "this information before combining."
         )
+
+
+class NoDataError(ValueError):
+    def __init__(self, source, key=None):
+        self.source = source
+        self.key = key
+
+    def __str__(self):
+        if self.key is not None:
+            return f'{self.source}, {self.key}'
+        else:
+            return self.source
```

Comment:
I'd give this a slightly more descriptive message, something like "No data in run/selection for source ABC, key DEF"

Reply:
I tried to model it after the builtin `KeyError` or `AttributeError`, which only state the thing missing and imply the reason by their type. But we can change it if you feel strongly about it.

Reply:
I see your point, without knowledge that `SourceNameError` and `PropertyNameError` exist one might not know that a `NoDataError` signals existance but emptiness. In addition, those errors are already fairly chatty.